skimage/future/detect_obj/cascade.pyx

```diff
@@ -0,0 +1,832 @@
+# cython: cdivision=True
+# cython: boundscheck=False
+# cython: nonecheck=False
+# cython: wraparound=False
+# distutils: language = c++
+
+import numpy as np
+cimport numpy as cnp
+cimport openmp
+from libc.stdlib cimport malloc, free
+from libcpp.vector cimport vector
+from skimage._shared.transform cimport integrate
+
+# Use our own implementation of the function instead of libc.math.
+# Otherwise, the compilation breaks on some compilers due to Cython
+# bug.
+from skimage._shared.interpolation cimport round, fmax, fmin
+
+from cython.parallel import prange
+from skimage.color import rgb2gray
+from skimage.transform import integral_image
+import xml.etree.ElementTree as ET
+from ...feature._texture cimport _multiblock_lbp
+import math
+
+
+# Struct for storing clusters of rectangles. As the rectangles are dynamically
+# added, the sum of row, col positions and width and heights are stored
+# with the count of rectangles that belong to this cluster. This way,
+# we don't have to store all the rectangles information as array
+# and the average can be easily computed in a constant time.
+cdef struct DetectionsCluster:
+
+    int r_sum
+    int c_sum
+    int width_sum
+    int height_sum
+    int count
+
+
+# Struct for storing a single detection.
+cdef struct Detection:
+
+    int r
+    int c
+    int width
+    int height
+
+
+# Struct for storing multi-block binary pattern position.
+cdef struct MBLBP:
```

Comment:
@stefanv Yes, we can. But the code this way is more readable.
I mean otherwise it can be confusing.
In case of MBLBP is mean the top-left rectangle coordinates and in case
of detection I just mean the rectangle that covers a detection region.